src/js/common/schemaform/widgets/CurrencyWidget.jsx

```diff
@@ -1,40 +1,54 @@
 import React from 'react';
 
 export default class CurrencyWidget extends React.Component {
+  constructor(props) {
+    super(props);
+    let value = props.value;
+    if (typeof value === 'number') {
+      value = value.toFixed(2);
+    }
+    this.state = {
+      value
+    };
+  }
+
   onBlur = () => {
     this.props.onBlur(this.props.id);
   }
 
   handleChange = (event) => {
     const val = event.target.value;
-    if (val === '') {
+    if (val === '' || typeof val === 'undefined') {
       this.props.onChange();
     } else {
-      // Sometimes we get a string back (e.g. when you enter 10.00)
-      // so we need to parse it to be sure
-      const parsed = parseFloat(val);
-      if (!isNaN(parsed)) {
-        this.props.onChange(parsed);
-      } else {
+      // Needs to look like a currency
+      if (!val.match(/^[\$0-9\.,]*$/)) {
```

Comment:
Works for me. I would say that we want to prevent anything that looks like an error or will parse in a surprising way. 

I added `$` to it, since I wanted to allow people to put that in there (accidentally). It's something that's easy to strip out.